app/src/main/java/org/mozilla/fenix/components/Core.kt

```diff
@@ -218,6 +219,18 @@ class Core(
                     .periodicallyInForeground(interval = 30, unit = TimeUnit.SECONDS)
                     .whenGoingToBackground()
                     .whenSessionsChange()
+
+                // Now that we have restored our previous state (if there's one) let's remove timed out tabs
+                if (!context.settings().manuallyCloseTabs) {
+                    store.state.tabs.filter {
+                        (System.currentTimeMillis() - it.lastAccess) > context.settings().getTabTimeout()
+                    }.forEach {
+                        val session = sessionManager.findSessionById(it.id)
+                        if (session != null) {
+                            sessionManager.remove(session)
+                        }
+                    }
+                }
```

Comment:
👍 Here's where we'd dispatch the new `RestoreCompleteAction` when ready. Then we can observe reliably that tabs are ready.